ee/tabby-webserver/src/handler.rs

```diff
@@ -104,13 +127,16 @@ impl WebserverHandle {
         api: Router,
         ui: Router,
         code: Arc<dyn CodeSearch>,
+        repository_access: Arc<dyn RepositoryAccess>,
         is_chat_enabled: bool,
         local_port: u16,
     ) -> (Router, Router) {
         let ctx = create_service_locator(
             self.logger(),
             code,
-            self.repository_service.clone(),
+            self.git_repository_service.clone(),
+            self.github_repository_service.clone(),
+            repository_access,
```

Comment:
```suggestion
            repository_access,
```

instead of add repository_access to service locator. Change the state of `ws_handler`, and pass `self.repository_access` all the way down.